pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/QuestionDao.java

```diff
@@ -410,15 +412,19 @@ default PicklistQuestion getPicklistQuestion(PicklistQuestionDto dto,
                 .map(rule -> (Rule<PicklistAnswer>) rule)
                 .collect(toList());
 
-        long timestamp = getHandle().attach(JdbiActivityInstance.class)
+        ActivityInstanceDto instanceDto = getHandle().attach(JdbiActivityInstance.class)
                 .getByActivityInstanceGuid(activityInstanceGuid)
-                .map(ActivityInstanceDto::getCreatedAtMillis)
+                .orElseThrow(() -> new DaoException("Could not find activity instance using guid " + activityInstanceGuid
+                        + " while getting picklist question " + dto.getStableId()));
+        ActivityVersionDto versionDto = ActivityDefStore.getInstance()
+                .findVersionDto(getHandle(), instanceDto.getActivityId(), instanceDto.getCreatedAtMillis())
                 .orElseThrow(() -> new DaoException(String.format(
-                        "Could not find activity instance with guid=%s while getting picklist question with id=%d and stableId=%s",
-                        activityInstanceGuid, dto.getId(), dto.getStableId())));
+                        "Could not find activity version for instance with guid=%s using"
+                                + " activityId=%d and createdAt=%d while getting picklist question %s",
+                        activityInstanceGuid, instanceDto.getActivityId(), instanceDto.getCreatedAtMillis(), dto.getStableId())));
 
         PicklistQuestionDao.GroupAndOptionDtos container = getPicklistQuestionDao()
-                .findOrderedGroupAndOptionDtos(dto.getId(), timestamp);
+                .findOrderedGroupAndOptionDtos(dto.getId(), versionDto.getRevStart());
```

Comment:
This method is used when calling the `Get Activity Instance` API, which is a separate codepath than loading activity definitions. We ensure we use the _activity definition timestamp_ rather than the instances' created_at timestamp, to avoid an explosion of cached items (since instance created_at timestamp is more-or-less unique!).